config/registry_image_pruner/image_pruner/prune_images.py

```diff
@@ -24,24 +24,42 @@
 ImageRepo = Dict[str, Any]
 
 
-def get_quay_repo(quay_token: str, namespace: str, name: str) -> ImageRepo:
-    api_url = f"{QUAY_API_URL}/repository/{namespace}/{name}"
-    request = Request(api_url, headers={
-        "Authorization": f"Bearer {quay_token}",
-    })
+def get_quay_tags(quay_token: str, namespace: str, name: str) -> List[ImageRepo]:
+    next_page = None
     resp: HTTPResponse
-    try:
+
+    all_tags = []
+    while True:
+        query_args = {"limit": 100, "onlyActiveTags": True}
+        if next_page is not None:
+            query_args["page"] = next_page
+
+        api_url = f"{QUAY_API_URL}/repository/{namespace}/{name}/tag/?{urlencode(query_args)}"
+        request = Request(api_url, headers={
+            "Authorization": f"Bearer {quay_token}",
+        })
+
         with urlopen(request) as resp:
             if resp.status != 200:
                 raise RuntimeError(resp.reason)
-            return json.loads(resp.read())
+            json_data = json.loads(resp.read())
 
-    except HTTPError as ex:
-        # ignore if not found
-        if ex.status != 404:
-            raise(ex)
+        tags = json_data.get("tags", [])
+        all_tags.extend(tags)
+
+        if not tags:
+            LOGGER.debug("No tags found.")
+            break
+
+        page = json_data.get("page", None)
+        additional = json_data.get("has_additional", False)
```

Comment:
It could be a condition for the while, so you can avoid `break` below.

Reply:
tbh this seems to be more readable, just to safe 2 lines of else & break